common/src/main/java/fr/esrf/TangoApi/HostInfo.java

```diff
@@ -30,159 +30,107 @@
 
 
-package fr.esrf.TangoApi;
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import fr.esrf.Tango.DevFailed;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.tango.utils.DevFailedUtils;
 
+import java.net.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 
 /**
- * This class manage the host information
- * - name
- * - address
+ * This class manage the host information obtained from {@link NetworkInterface#getInetAddresses()}
+ * - name -- last found NI name
+ * - address -- last found NI address
+ * - addresses -- list of NI addresses
  *
  * @author verdier
  */
-
-import fr.esrf.Tango.DevFailed;
-import fr.esrf.TangoDs.Except;
-
-import java.net.InetAddress;
-import java.net.NetworkInterface;
-import java.net.SocketException;
-import java.util.Enumeration;
-import java.util.StringTokenizer;
-import java.util.Vector;
-
-
 public class HostInfo {
-	private static String name = null;
-	private static String address = null;
-	private static Vector<String> addresses = new Vector<String>();
-
-	//===============================================================
-	//===============================================================
-	private HostInfo() throws DevFailed {
-		String env = System.getenv("TraceAddresses");
-		boolean trace = (env != null && env.equals("true"));
-		try {
-
-			Enumeration<NetworkInterface> enet = NetworkInterface.getNetworkInterfaces();
-			while (enet.hasMoreElements()) {
-				NetworkInterface net = enet.nextElement();
-
-				//  To do not have 127.x.x.x
-				if (net.isLoopback())
-					continue;
-
-				Enumeration<InetAddress> eaddr = net.getInetAddresses();
-				if (trace) System.out.println("----------------- " + net.getName() + " --------------------");
-				while (eaddr.hasMoreElements()) {
-					InetAddress inetAddress = eaddr.nextElement();
-					if (trace) {
-                        System.out.println("getCanonicalHostName(): " + inetAddress.getCanonicalHostName());
-                        System.out.println("getHostName():          " + inetAddress.getHostName());
-                        System.out.println("getHostAddress():       " + inetAddress.getHostAddress());
+    /**
+     * A jacorb system property for IP address on multi-homed host
+     */
+    public static final String OAI_ADDR = System.getProperty("OAIAddr");
+    private static final Logger LOGGER = LoggerFactory.getLogger(HostInfo.class);
+
+    //===============================================================
+    //===============================================================
+    private HostInfo() {
+    }
+
+    /**
+     * @return all Ip (4&6) available on this host
+     * @throws DevFailed
+     */
+    public static Iterable<String> getIpAddresses() throws DevFailed {
+        List<String> result;
+        if (OAI_ADDR != null && !OAI_ADDR.isEmpty()) {
```

Comment:
I am not sure about this one. I would return all possible addresses in this method. This check can be moved to methods related to single address, i.e. _getAddress_ etc